lasy/profiles/transverse/super_gaussian_profile.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from .transverse_profile import TransverseProfile
+
+class SuperGaussianTransverseProfile(TransverseProfile):
+    """
+    Derived class for the analytic profile of a super-Gaussian laser pulse.
+    """
+
+    def __init__(self, w0, n_order):
+        """
+        Defines a super-Gaussian transverse envelope.
+
+        More precisely, the transverse envelope corresponds to:
+
+        .. math::
+
+            \\mathcal{T}(x, y) = \\exp\\left( -\\left({\\frac{{x^2 + y^2}}{w_0^2}}\\right)^{\\dfrac{n}{2}} \\right)
+
+        Parameters
+        ----------
+        w0: float (in meter)
+            The waist of the laser pulse, i.e. :math:`w_0` in the above formula.
+
+        n_order: float (in meter)
+            The shape parameter of the super-gaussian function, i.e. :math:`n` in the above formula.
+            If :math:`n=2` the super-Gaussian becomes a standard Gaussian function.
+            If :math:`n=1` the super-Gaussian becomes a Laplace function.
+        """
+        self.w0 = w0
```

Comment:
As (pretty neatly) suggested by codeQL, I think we should call the base Classe's `__init__`
```suggestion
        super().__init__()
        self.w0 = w0
```

Reply:
good idea!